benches/bench_recovery.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2017-present, PingCAP, Inc. Licensed under Apache-2.0.
+
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use raft::eraftpb::Entry;
+use raft_engine::ReadableSize;
+use raft_engine::{Config as EngineConfig, LogBatch, MessageExt, RaftLogEngine, Result};
+use rand::{Rng, SeedableRng};
+use std::collections::HashMap;
+use std::fmt;
+use std::path::PathBuf;
+use tempfile::TempDir;
+
+extern crate libc;
+
+type Engine = RaftLogEngine<MessageExtTyped>;
+
+#[derive(Clone)]
+struct MessageExtTyped;
+impl MessageExt for MessageExtTyped {
+    type Entry = Entry;
+
+    fn index(entry: &Entry) -> u64 {
+        entry.index
+    }
+}
+
+struct Config {
+    total_size: ReadableSize,
+    region_count: u64,
+    batch_size: ReadableSize,
+    item_size: ReadableSize,
+    entry_size: ReadableSize,
+    batch_compression_threshold: ReadableSize,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            total_size: ReadableSize::gb(1),
+            region_count: 100,
+            batch_size: ReadableSize::mb(1),
+            item_size: ReadableSize::kb(1),
+            entry_size: ReadableSize(256),
+            batch_compression_threshold: ReadableSize(0),
+        }
+    }
+}
+
+impl fmt::Display for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{} [region-count: {}][batch-size: {}][item-size: {}][entry-size: {}][batch-compression-threshold: {}]",
+        self.total_size,
+        self.region_count,
+        self.batch_size,
+        self.item_size,
+        self.entry_size,
+        self.batch_compression_threshold
+    )
```

Comment:
```suggestion
        write!(f, "{} [region-count: {}][batch-size: {}][item-size: {}][entry-size: {}][batch-compression-threshold: {}]",
            self.total_size,
            self.region_count,
            self.batch_size,
            self.item_size,
            self.entry_size,
            self.batch_compression_threshold
        )
```

Reply:
It weird that my formatter tells no difference on both.